pymongo/pool.py

```diff
@@ -1794,6 +1806,9 @@ def checkin(self, conn: Connection) -> None:
         conn.pinned_cursor = False
         self.__pinned_sockets.discard(conn)
         listeners = self.opts._event_listeners
+        with self.lock:
+            if conn.cancel_context in self.active_conns_context:
```

Comment:
set.remove already handles this case so the if statement can be removed.

Reply:
Turns out `set.discard` is the method we're looking for, good catch!